anndata/_core/views.py

```diff
@@ -308,15 +308,19 @@ def as_view_awkarray(array, view_args):
         # possible strategies to stack behaviors.
         # A better solution might be based on xarray-style "attrs", once this is implemented
         # https://github.com/scikit-hep/awkward/issues/1391#issuecomment-1412297114
-        if type(array).__name__ != "Array":
+        #
+        # Allow only Arrays that have no record behavior attached, or with __record__ explicitly
+        # set to None
+        if ak.parameters(array).get("__record__", None) is not None:
```

Comment:
(Record) names can be set at arbitrary depths. I'd suggest checking recursively using our L2 (public-developer) API:
```suggestion
        if array.layout.purelist_parameter("__record__") is not None:
```

Reply:
[This setitem](https://github.com/scverse/anndata/blob/b6406179f50ce69d84112a5426d206a203a1181d/anndata/_core/views.py#L323) is responsible for registering the behavior class for lists containing records, rather than 1D arrays of records.